app/controllers/users_controller.rb

```diff
@@ -16,20 +16,38 @@ def create
 
   def show
     require_user
+    @user = current_user
+  end
+
+  def edit_password 
+  end
+
+  def update
+    if current_user.update(user_params)
+      if params[:user][:password]
```

Comment:
functionality looks great, our update method is getting a little long, probably a candidate for refactoring